pkg/storage/tsdb/block/fetcher.go

```diff
@@ -307,7 +317,18 @@ func (f *MetaFetcher) fetchMetadata(ctx context.Context, excludeMarkedForDeletio
 		ch  = make(chan ulid.ULID, f.concurrency)
 		mtx sync.Mutex
 	)
-	level.Debug(f.logger).Log("msg", "fetching meta data", "concurrency", f.concurrency)
+
+	level.Debug(f.logger).Log("msg", "fetching meta data", "concurrency", f.concurrency, "max-lookback", f.maxLookback)
+	var minAllowedBlockID ulid.ULID
+	if f.maxLookback.Milliseconds() > 0 {
+		var err error
+		minAllowedBlockID, err = ulid.New(
```

Comment:
The way we are constructing a ULID that represents a point in time - and then comparing that to items in the list in order to filter by age is deserving of a handrail "why" comment.